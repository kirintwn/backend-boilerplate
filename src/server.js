import Koa from 'koa';
import koaLogger from 'koa-logger';
import bodyParser from 'koa-body';
import router from './routes';

const server = new Koa();
server.proxy = true;
server.use(bodyParser({ multipart: true }));

server.use(koaLogger());

server.use(router.routes());

export default server;
