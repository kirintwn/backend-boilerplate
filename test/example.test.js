/* eslint-disable no-console */
/* eslint-disable no-await-in-loop */
/* eslint-disable no-unused-vars */
import 'core-js/stable';
import 'regenerator-runtime/runtime';
import supertest from 'supertest';
import server from '../src';

const { PORT = 3001 } = process.env;
let listener;
let request;

beforeAll(async () => {
  listener = server.listen(PORT);
  request = supertest(listener);
});

afterAll(async () => {
  await listener.close();
});

beforeEach(async () => {
  // Add Smth
});

afterEach(async () => {
  // Add Smth
});

describe('test koa server', () => {
  test('should respond as expected', async () => {
    const response = await request.get('/test');
    expect(response.status).toEqual(200);
    expect(response.body.data).toMatchSnapshot();
  });
});
