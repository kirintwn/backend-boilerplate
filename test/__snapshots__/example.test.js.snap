// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test koa server should respond as expected 1`] = `"it works!"`;
